### ISSUE TITLE
Panic when calling the gas histogram RPC

### ISSUE THREAD

**Issue by chevdor** [ethcore/parity#3146]
```
thread '<unnamed>' panicked at 'index out of bounds: the len is 1762 but the index is 1762', util/src/stats.rs:45
stack backtrace:
   1:        0x10c638729 - std::sys::backtrace::tracing::imp::write::hd4b54a4a2078cb15
   2:        0x10c63f610 - std::panicking::default_hook::_{{closure}}::h51a5ee7ba6a9fcef
   3:        0x10c63e470 - std::panicking::default_hook::hf823fce261e27590
   4:        0x10c63eaf6 - std::panicking::rust_panic_with_hook::h8d486474663979b9
   5:        0x10c63e944 - std::panicking::begin_panic::h72862f004a4942ab
   6:        0x10c63e8a2 - std::panicking::begin_panic_fmt::hdc424a357d9142e1
   7:        0x10c63e807 - rust_begin_unwind
   8:        0x10c66c5e0 - core::panicking::panic_fmt::h6b06f78ae7f9dd57
   9:        0x10c66c558 - core::panicking::panic_bounds_check::hf8612aab39b77581
  10:        0x10c3cc9b2 - ethcore_util::stats::Histogram::new::h30e90d26d37f8494
  11:        0x10bb96423 - ethcore::client::traits::BlockChainClient::gas_price_histogram::he921adc7939479dd
  12:        0x10baae74c - _<ethcore_rpc..v1..impls..ethcore..EthcoreClient<C, M, S, F> as ethcore_rpc..v1..traits..ethcore..Ethcore>::gas_price_histogram::h78ca106d7d70952c
  13:        0x10ba15da4 - _<jsonrpc_core..io..DelegateMethod<T, F> as jsonrpc_core..commander..MethodCommand>::execute::h0de3357af2a00d92
  14:        0x10be98254 - jsonrpc_core::commander::Commander::execute_method::hbb03a32ee296e872
  15:        0x10be993a1 - jsonrpc_core::request_handler::RequestHandler::handle_call::hc855c8986a6ce396
  16:        0x10be98825 - jsonrpc_core::request_handler::RequestHandler::handle_request::hfc62e480a725dce1
  17:        0x10be9b7aa - jsonrpc_core::io::IoHandler::handle_request::h32211be56e10a1ce
  18:        0x10bd15cb5 - _<ethcore_signer..ws_server..session..Session as ws..handler..Handler>::on_message::ha07f33f46d89f0cd
  19:        0x10bcf3728 - _<ws..connection..Connection<H>>::read_frames::h30f37e99ad10d979
  20:        0x10bcf9605 - _<ws..connection..Connection<H>>::read::he3b7d4358d4b66d9
  21:        0x10bd08663 - _<ws..io..Handler<F> as mio..handler..Handler>::ready::h4f6d5a629d43349a
  22:        0x10bceb5b1 - std::panicking::try::do_call::ha29ac1f3779b0e92
  23:        0x10c63fbfa - __rust_maybe_catch_panic
  24:        0x10bd06c7a - _<F as alloc..boxed..FnBox<A>>::call_box::h7f2daa29526a36d1
  25:        0x10c63da95 - std::sys::thread::Thread::new::thread_start::h57f688c224d4fa4d
  26:     0x7fffc289caaa - _pthread_body
  27:     0x7fffc289c9f6 - _pthread_start
2016-11-03 18:23:05  Finishing work, please wait...
2016-11-03 18:23:05  Signer: Broadcaster was not closed cleanly. Details: WS Error <Queue(NotifyError::Closed(..))>
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Any', ../src/libcore/result.rs:788
stack backtrace:
   1:        0x10c638729 - std::sys::backtrace::tracing::imp::write::hd4b54a4a2078cb15
   2:        0x10c63f610 - std::panicking::default_hook::_{{closure}}::h51a5ee7ba6a9fcef
   3:        0x10c63e470 - std::panicking::default_hook::hf823fce261e27590
   4:        0x10c63eaf6 - std::panicking::rust_panic_with_hook::h8d486474663979b9
   5:        0x10c63e944 - std::panicking::begin_panic::h72862f004a4942ab
   6:        0x10c63e8a2 - std::panicking::begin_panic_fmt::hdc424a357d9142e1
   7:        0x10c63e807 - rust_begin_unwind
   8:        0x10c66c5e0 - core::panicking::panic_fmt::h6b06f78ae7f9dd57
   9:        0x10bd04051 - core::result::unwrap_failed::h6719666e66463f8c
  10:        0x10bd1ceb2 - _<ethcore_signer..ws_server..Server as core..ops..Drop>::drop::ha91df67f481ae0c4
  11:        0x10bc30de8 - parity::run::execute::h8f6325ecb3490d28
  12:        0x10bc3ad58 - parity::execute::hd77573b1fe96f12c
  13:        0x10bc4551d - parity::main::h0d17049d06fcd993
  14:        0x10c63fbfa - __rust_maybe_catch_panic
  15:        0x10c63df3e - std::rt::lang_start::hca48e539ce72a288
```

**Comment by rphmeier** [comment]
Fixed in #3110

**Comment by chevdor** [comment]
@rphmeier Are you sure? what you reference has been merged (10) hours ago. I pulled a few minutes ago and I see the issue.

**Comment by gavofyork** [comment]
are you certain you rebuilt and are running the latest version? could you post `parity --version`?

**Comment by chevdor** [comment]
I am on master:
`Parity/v1.4.0-unstable-**3cb355d**-20161103/x86_64-macos/rustc1.12.1`

**Comment by rphmeier** [comment]
@chevdor my mistake, I thought it was the same error.

**Comment by keorn** [comment]
Closed by #3150
